tests/operators/test_offres_exp_approx.py

```diff
@@ -0,0 +1,169 @@
+"""Test off-resonance spatial coefficient and temporal interpolator estimation."""
+
+import math
+
+import numpy as np
+import numpy.testing as npt
+
+import pytest
+
+import mrinufft
+from mrinufft.operators.base import CUPY_AVAILABLE, AUTOGRAD_AVAILABLE
+from mrinufft._utils import get_array_module
+
+from helpers import to_interface, from_interface
+
+
+def calculate_true_offresonance_term(fieldmap, t):
+    """Calculate non-approximate off-resonance modulation term."""
+    xp = get_array_module(fieldmap)
+    t = xp.asarray(t)
+    if xp.__name__ == "torch":
+        t = t.to(device=fieldmap.device)
+    arg = t * fieldmap[..., None]
+    arg = arg[None, ...].swapaxes(0, -1)[..., 0]
+    return xp.exp(-arg)
+
+
+def calculate_approx_offresonance_term(B, C):
+    """Calculate approximate off-resonance modulation term."""
+    field_term = 0.0
+    for n in range(B.shape[0]):
+        tmp = B[n] * C[n][..., None]
+        tmp = tmp[None, ...].swapaxes(0, -1)[..., 0]
+        field_term += tmp
+    return field_term
+
+
+def _make_b0map(ndim, npix=64):
+    """Make B0 map (units: Hz)."""
+    shape = ndim * [npix]
+    if ndim == 2:
+        mask, fieldmap = _make_disk(shape)
+    elif ndim == 3:
+        mask, fieldmap = _make_sphere(shape)
+    fieldmap *= mask
+    fieldmap = 600 * fieldmap / fieldmap.max() - 300  # Hz
+    fieldmap *= mask
+    return fieldmap.astype(np.float32), mask
+
+
+def _make_t2smap(ndim, npix=64):
+    """Make T2* map (units: ms)."""
+    shape = ndim * [npix]
+    if ndim == 2:
+        mask, _ = _make_disk(shape)
+    elif ndim == 3:
+        mask, _ = _make_sphere(shape)
+    fieldmap = 15.0 * mask  # ms
+    return fieldmap.astype(np.float32), mask
+
+
+def _make_disk(shape, frac_radius=0.3):
+    """Make circular binary mask."""
+    ny, nx = shape
+    yy, xx = np.mgrid[:ny, :nx]
+    yy, xx = yy - ny // 2, xx - nx // 2
+    yy, xx = yy / ny, xx / nx
+    rr = (xx**2 + yy**2) ** 0.5
+    return rr < frac_radius, rr
+
+
+def _make_sphere(shape, frac_radius=0.3):
+    """Make spherical binary mask."""
+    nz, ny, nx = shape
+    zz, yy, xx = np.mgrid[:nz, :ny, :nx]
+    zz, yy, xx = zz - nz // 2, yy - ny // 2, xx - nx // 2
+    zz, yy, xx = zz / nz, yy / ny, xx / nx
+    rr = (xx**2 + yy**2 + zz**2) ** 0.5
+    return rr < frac_radius, rr
+
+
+# Parameter combinations (ndims, backend, device)
+params = [(2, "cpu", None), (3, "cpu", None)]
+
+if CUPY_AVAILABLE:
+    params.extend([(2, "gpu", None), (3, "gpu", None)])
+
+if AUTOGRAD_AVAILABLE:
+    params.extend([(2, "torch", "cpu"), (3, "torch", "cpu")])
+
+if AUTOGRAD_AVAILABLE and CUPY_AVAILABLE:
+    params.extend([(2, "torch", "gpu"), (3, "torch", "gpu")])
+
+
+@pytest.fixture(scope="module", params=params)
+def map_fixture(request):
```

Comment:
A few suggestions:
- Just name it b0_map, the fixture suffix is redundant.
- Use `params_array_interface` decorator
- parametrize the fixture/TestCase directly (see the `operator` one in `test_batch.py` or `test_interface.py` for instance